src/app/static/src/app/components/Warning.vue

```diff
@@ -0,0 +1,173 @@
+<template>
+    <div v-if="warnings.length > 0">
+        <h4 class="alert-heading pt-2">
+            <alert-triangle-icon size="1.5x" class="custom-class mr-1 mb-1"></alert-triangle-icon>
+            <span v-translate="headerText(origin)"></span>
+        </h4>
+        <!-- The outer styled divs are the visible window of warnings and inside them are the 
+        full window of warnings (the warningBox), which is always rendered as html but with 
+        its overflow hidden by the outer box. The size of the outer box dynamically adjusts 
+        depending on the size of the warningBox and whether the user has clicked show more or less. -->
+        <div :style="{ overflowY: 'hidden', height: `${this.renderedBoxHeight}px` }">
+            <ul class="mb-0" ref="warningBox" id="warningBox">
+                <li v-for="warning in warnings" :key="warning.text"><div :style="lineStyling">{{ warning.text }}</div></li>
```

Comment:
I've read mixed messages about whether it is ok to put a `div` inside a `li` but I needed to do this to prevent the bullets from disappearing after applying the style modifications.